### PR TITLE
HP9825: added 9825T variant

### PR DESCRIPTION
**Pull request by fulivi**
Hi,
in this PR I'm adding the "T" variant of hp9825.
I'm going to send soon the new ROM set to code email address.
Thanks.
--F.Ulivi

### CODE REVIEW

File: src/devices/cpu/hphybrid/hphybrid.cpp [hunk 1]
```diff
@@ -241,6 +243,11 @@ void hp_hybrid_cpu_device::device_start()
 	set_icountptr(m_icount);
 
 	m_pa_changed_func.resolve_safe();
+	m_opcode_func.resolve();
+	m_stm_func.resolve();
+	// Cache active state of m_stm_func & m_opcode_func
+	m_opcode_func_defd = bool(m_opcode_func);
+	m_stm_func_defd = bool(m_stm_func);
```
Comment (cuavas) [inline src/devices/cpu/hphybrid/hphybrid.cpp]: This is a bad pattern - the check on the devcb is fast anyway, but now you've got stuff that can potentially get out of sync.  For `m_opcode_func` you can just use `resolve_safe` and call it without the check.

Comment (fulivi) [reply src/devices/cpu/hphybrid/hphybrid.cpp]: Yes, I know, I don't like it very much either. I started in the canonical way (`resolve_safe` etc) but then I noticed it has some speed penalty on 9845 emulation (two hybrid processors there). A rough benchmark shows that `resolve_safe` has the emulation running at 800/900% at maximum speed (PgDown pressed) whereas the bool cache gets to around 1000/1100%.
Should I return to `resolve_safe` in your opinion?

Comment (cuavas) [reply src/devices/cpu/hphybrid/hphybrid.cpp]: When it's already running that much faster than realtime, the speed hit is immaterial.

Comment (rb6502) [reply src/devices/cpu/hphybrid/hphybrid.cpp]: I don't know how resolve_safe is implemented, but returning a constant shouldn't be 1-2 complete emulations of the machine slower than calling through a delegate.  We routinely praise Ted and Ryan for finding lesser speedups in other drivers, after all.

**Pull request merged**